src/kixi/datastore/metadatastore/elasticsearch.clj

```diff
@@ -0,0 +1,239 @@
+  [f]
+  (fn mapper [m]
+    (cond
+      (map? m) (zipmap (map f (keys m))
+                       (map (fn [v]
+                              (cond  
+                                (map? v) (mapper v)
+                                (list? v) (map mapper v) 
+                                (vector? v) (mapv mapper v)
+                                (seq? v) (map mapper v) 
+                                (symbol? v) (name v)
+                                (keyword? v) (f v)
+                                :else v))
+                            (vals m)))
+      (keyword? m) (f m)
+      :else m)))
+
+(def all-keys->es-format
+  (map-all-keys kw->es-format))
+
+(def all-keys->kw
+  (map-all-keys es-format->kw))
+
+(def string-stored-not_analyzed
+  {:type "string" 
+   :store "yes"
+   :index "not_analyzed"})
+
+(defn build-index
+  [conn]
+  (let [mapping-types {doc-type {:properties 
+                                 (all-keys->es-format
+                                  {::ms/id string-stored-not_analyzed
+                                   ::ms/type string-stored-not_analyzed
+                                   ::ms/name {:type "string"
+                                              :store "yes"}
+                                   ::ss/id string-stored-not_analyzed
+                                   ::ms/provenance {:type "nested"
+                                                    :properties {::ms/source string-stored-not_analyzed
+                                                                 :kixi.user/id string-stored-not_analyzed
+                                                                 ::ms/pieces-count {:type "integer"}
+                                                                 ::ms/parent-id string-stored-not_analyzed}}
+                                   ::ms/segmentation {:type "nested"
+                                                      :properties {::seg/type string-stored-not_analyzed
+                                                                   ::seg/line-count {:type "integer"}
+                                                                   ::seg/value string-stored-not_analyzed}}
+                                   ::ms/sharing {:type "nested"
+                                                 :properties (zipmap ms/activities
+                                                                     (repeat string-stored-not_analyzed))}})}}]
+    (esi/create conn index-name {:mappings mapping-types
+                                 :settings {}})))
+
+(defn ensure-index
+  [conn]
+  (when-not (esi/exists? conn index-name)
+    (build-index conn)))
+
+(defn- get-document
+  [conn id]
+  (esd/get conn
+           index-name 
+           doc-type 
+           id
+           {:preference "_primary"}))
+
+(s/fdef update-metadata-processor
+        :args (s/cat :conn #(instance? clojurewerkz.elastisch.rest.Connection %)
+                     :update-req ::cs/file-metadata-updated))
```

Comment:
Well the result is completely unused at the moment, and they don't do anything unless you're generating the tests. Which for these functions seems very unlikely!